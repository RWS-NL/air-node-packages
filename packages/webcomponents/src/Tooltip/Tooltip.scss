@import '../styles/vars';

.tooltip {
  background-color: $white !important;
  border-radius: 0 !important;
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 20%), 0 1px 1px 0 rgb(0 0 0 / 14%), 0 2px 1px -1px rgb(0 0 0 / 12%) !important;
  color: rgb(0 0 0 / 87%) !important;
  font-size: 0.8rem !important;
}

.box {
  padding: 8px;
}
