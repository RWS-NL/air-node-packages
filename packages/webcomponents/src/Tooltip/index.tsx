import Box from '@material-ui/core/Box';
import MUITooltip, { TooltipProps as MUITooltipProps } from '@material-ui/core/Tooltip';
import { CSSProperties } from '@material-ui/styles';
import classnames from 'classnames';
import React, { memo } from 'react';
import css from './Tooltip.scss';

export interface TooltipProps extends MUITooltipProps {
  /** Data-qa tag to apply to the tooltip */
  'data-qa'?: string;
  /** Custom CSS classes to pass to the tooltip */
  customclasses?: string | string[];
  /** Any additional CSSProperties to pass to the component */
  style?: CSSProperties;
}

/**
 * Constructs a tooltip using pre-defined Rijkswaterstaat styling
 * @param param Props to pass to the Tooltip
 * @example
 * ```jsx
 * <Tooltip title='Tooltip Content' placement='top'>
 *   <IconButton>
 *     <CloudDownload color='primary' />
 *   </IconButton>
 * </Tooltip>
 * ```
 */
export const Tooltip = memo(
  ({ 'data-qa': qaTag, title, placement, enterDelay, customclasses, style, children, ...props }: TooltipProps) => (
    <MUITooltip
      {...props}
      title={
        <Box data-qa={qaTag} className={css.box}>
          {title}
        </Box>
      }
      placement={placement || 'top'}
      enterDelay={enterDelay || 300}
      classes={{ tooltip: classnames(css.tooltip, customclasses), popper: css.popper }}
      style={style}
    >
      {children}
    </MUITooltip>
  )
);

export default Tooltip;
