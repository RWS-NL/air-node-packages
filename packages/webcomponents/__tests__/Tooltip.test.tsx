import { shallow } from 'enzyme';
import React from 'react';
import Tooltip from '../src/Tooltip/Tooltip';

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const tooltip = shallow(<Tooltip title='SnapTitle'><span>Snappy Text</span></Tooltip>);
    expect(tooltip).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const simpleTooltip = shallow(
      <Tooltip title='SnapTitle' data-qa='snap-qa' style={{ borderLeftColor: 'red' }}>
        <span>Snappy Text</span>
      </Tooltip>
    );
    expect(simpleTooltip).toMatchSnapshot();
  });
});