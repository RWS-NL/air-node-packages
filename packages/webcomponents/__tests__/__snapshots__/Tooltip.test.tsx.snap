// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Optional Props 1`] = `
<WithStyles(ForwardRef(Tooltip))
  classes={
    Object {
      "popper": "popper",
      "tooltip": "tooltip",
    }
  }
  enterDelay={300}
  placement="top"
  style={
    Object {
      "borderLeftColor": "red",
    }
  }
  title={
    <ForwardRef(StyledComponent)
      className="box"
      data-qa="snap-qa"
    >
      SnapTitle
    </ForwardRef(StyledComponent)>
  }
>
  <span>
    Snappy Text
  </span>
</WithStyles(ForwardRef(Tooltip))>
`;

exports[`Snapshot Testing Required Props 1`] = `
<WithStyles(ForwardRef(Tooltip))
  classes={
    Object {
      "popper": "popper",
      "tooltip": "tooltip",
    }
  }
  enterDelay={300}
  placement="top"
  title={
    <ForwardRef(StyledComponent)
      className="box"
    >
      SnapTitle
    </ForwardRef(StyledComponent)>
  }
>
  <span>
    Snappy Text
  </span>
</WithStyles(ForwardRef(Tooltip))>
`;
